homeassistant/components/solaredge/__init__.py

```diff
@@ -1,18 +1,49 @@
-"""The solaredge integration."""
+"""The SolarEdge integration."""
 from __future__ import annotations
 
+from requests.exceptions import ConnectTimeout, HTTPError
+from solaredge import Solaredge
+
 from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_API_KEY
 from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
 import homeassistant.helpers.config_validation as cv
 
-from .const import DOMAIN
+from .const import CONF_SITE_ID, DATA_API_CLIENT, DOMAIN, LOGGER
 
 CONFIG_SCHEMA = cv.deprecated(DOMAIN)
 
+PLATFORMS = ["sensor"]
+
 
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
-    """Load the saved entities."""
-    hass.async_create_task(
-        hass.config_entries.async_forward_entry_setup(entry, "sensor")
-    )
+    """Set up SolarEdge from a config entry."""
+    api = Solaredge(entry.data[CONF_API_KEY])
+
+    try:
+        response = await hass.async_add_executor_job(
+            api.get_details, entry.data[CONF_SITE_ID]
+        )
+        if response["details"]["status"].lower() != "active":
```

Comment:
We can move this part out of the try... except block to keep that minimal

Reply:
Quick improvement, done 👍 Thanks!